File: 55-jump-game/jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let maxIdx = 0;

    for (let i = 0; i < nums.length; i++) {
        if (i > maxIdx) {
            return false
        };

        maxIdx = Math.max(maxIdx, i + nums[i]);
        
        if (maxIdx > nums.length - 1) {
            return true
        }
    }
    return true;
};

File: 55-jump-game/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxIdx = 0

        for i in range(0, len(nums)):
            if i > maxIdx:
                return False
            
            maxIdx = max(maxIdx, i + nums[i])

            if maxIdx > len(nums) - 1:
                return True
        
        return True
